### PR TITLE
[Runtime Metrics] Enable runtime metrics with memory profiling

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Enable runtime metrics when memory profiling is enabled.

## What

- consider memory profiling configuration when enabling runtime metrics
- update `metrics.md`

## Tests

Included in PR.

### CODE REVIEW

File: tracer/test/Datadog.Trace.Tests/TracerSettingsTests.cs [hunk 1]
```diff
@@ -106,6 +106,24 @@ public void ReplaceLocalhost(string original, string expected)
             Assert.Equal(expected, tracerSettings.ExporterSettings.AgentUri.ToString());
         }
 
+        [Theory]
+        [InlineData("1", "0", true)]
```
Comment (Kielek) [inline tracer/test/Datadog.Trace.Tests/TracerSettingsTests.cs:110]: I think that you have missed `null` as correct value for the test cases.
I think that, null, 1  => true, but "0", "1" => false.

File: docs/metrics.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ to Splunk Observability Cloud.
 To enable runtime metrics, set the `SIGNALFX_RUNTIME_METRICS_ENABLED` environment
 variable to `true` for your .NET process.
 
+> Runtime metrics are automatically enabled when [AlwaysOn memory profiling](internal/memory-profiling.md) is enabled.
```
Comment (theletterf) [inline docs/metrics.md:13]: What happens if you set `SIGNALFX_RUNTIME_METRICS_ENABLED` to `false` and enable memory profiling? Unlikely, but possible. Profiling settings overrides Runtime metrics? We might want to add this.

**Review by theletterf — APPROVED** [review]

File: docs/metrics.md (hunk 2 above)
Comment (lachmatt) [reply docs/metrics.md]: > Profiling settings overrides Runtime metrics

This is expected behavior as far as I know. Updated the docs.

File: tracer/test/Datadog.Trace.Tests/TracerSettingsTests.cs (hunk 1 above)
Comment (lachmatt) [reply tracer/test/Datadog.Trace.Tests/TracerSettingsTests.cs:110]: Added additional test cases. Current implementation seems to be aligned with other languages and is expected as far as I know. I will confirm that.

**Review by Kielek — APPROVED** [review]

**Pull request merged**